client.go

```diff
@@ -423,3 +427,62 @@ func (c *Client) String() string {
 func (c *Client) setHeader(key, val string) {
 	c.headers[key] = val
 }
+
+func (c *Client) getEventSource(fql *Query, opts ...QueryOptFn) (*EventSource, error) {
+	res, err := c.Query(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	token, ok := res.Data.(EventSource)
+	if !ok {
+		return nil, fmt.Errorf("query should return a fauna.EventSource but got %T", res.Data)
+	}
+
+	return &token, nil
+}
+
+// FeedFromQuery opens an event feed from the event source returned by the [fauna.Query].
+func (c *Client) FeedFromQuery(fql *Query, opts ...QueryOptFn) (*EventFeed, error) {
+	token, err := c.getEventSource(fql, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return newEventFeed(c, *token)
+}
+
+// FeedFromQueryWithStartTime initiates an event from the event source returned by the [fauna.Query] with custom options
+func (c *Client) FeedFromQueryWithStartTime(fql *Query, start FeedStartFn, opts ...QueryOptFn) (*EventFeed, error) {
```

Comment:
I thought you meant you'd use functions as an implementation detail!

The API should probably take an int64 directly and wrap it in a fn to give to newEventFeed, or accept the concrete fn instead of the interface.